File: C++/binary_search.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void)
{
	int arr[] = { 2, 3, 4, 10, 40 };
	int x = 10;
	int n = sizeof(arr) / sizeof(arr[0]);
	
	int l=0,r=n-1,flag=0;
	
	while(l<=r)
	{
		int m=l+(r-l)/2;
		if(arr[m]==x)
		{	
			cout<<"Element Found at index : "<<m<<endl;
			flag=1;
		}
		else if(arr[m]>x)
			r=m-1;
		else
			l=m+1;
	}
	
	if(!flag)
		cout<<"The element does not exist"<<endl;
}

File: C#/BinarySearch.cs
namespace DefaultNamespace;

public class BinarySearch
{
    // Returns index of x if it is present in arr[l..r], else return -1
    private static int binarySearch(int[] arr, int l, int r, int x)
    {
        if (r >= l)
        {
            var mid = l + (r - l) / 2;
            // If the element is in the middle.
            if (arr[mid] == x)
                return mid;
            // If element is lower than the mid point, then it can only be present in left subarray.
            if (arr[mid] > x)
                return binarySearch(arr, l, mid - 1, x);
            // Else the element can only be present in right subarray
            return binarySearch(arr, mid + 1, r, x);
        }
        // Return -1 if the index was not present within the array.
        return -1;
    }
    // Method to test the function above
    public static void Main()
    {
        int[] arr = { 3, 4, 45, 11, 41 };
        var n = arr.Length;
        var x = 11;
        var result = binarySearch(arr, 0, n - 1, x);
        if (result == -1)
            Console.WriteLine("Element not present");
        else
            Console.WriteLine($"Element found at index {result}");
    }
}